client/src/components/ui/inputs.tsx

```diff
@@ -0,0 +1,348 @@
+    setIsSelected(false);
+    onChange({ ...e, target: { ...e.target, value: fixedValue } });
+  };
+
+  return (
+    <div
+      className={`${isSelected ? "bg-opacity-0" : "bg-opacity-5"} ${label ? textStyleWithLabel : textStyleNoLabel}`}
+    >
+      {label && <InputLabel label={label} id={id} />}
+      <div className="item-center flex size-full flex-row">
+        <span className="size-4">$</span>
+        <input
+          id={id}
+          type="number"
+          value={value}
+          placeholder={placeholder}
+          onChange={handleOnChange}
+          onFocus={() => setIsSelected(true)}
+          onBlur={handleOnBlur}
+          className={
+            `${valueStyle} w-full` + // Align right spinner to end of input
+            (valueChanged
+              ? " text-penni-text-regular-light-mode"
+              : " text-penni-text-tertiary-light-mode")
+          }
+        />
+      </div>
+    </div>
+  );
+}
+
+/**
+ * ParagraphInput Component
+ *
+ * @param props - The properties for the ParagraphInput component.
+ * @param props.value - The current text value of the textarea, tracked by `react.setState()`.
+ * @param props.onChange - Called when the text value changes, should update `value`.
+ * @param props.label - (*Optional*) Label for the field, displayed above the textarea.
+ * @param props.placeholder - (*Optional*) Placeholder text for the textarea.
+ *
+ * @returns Textarea component for entering paragraphs of text.
+ *
+ * @example
+ * // Example usage:
+ * const [paragraphValue, setParagraphValue] = useState('');
+ * <ParagraphInput
+ *   value={paragraphValue}
+ *   onChange={(e) => setParagraphValue(e.target.value)}
+ *   label="Enter text"
+ *   placeholder="Type here..."
+ * />
+ */
+function ParagraphInput({
+  value,
+  onChange,
+  label,
+  placeholder,
+}: FreeTextInputProps) {
+  const [valueChanged, setValueChanged] = useState(false); // show lighter grey if not changed
+  const [id] = useState(uniqueId());
+  const [isSelected, setIsSelected] = useState(false);
+
+  return (
+    <div
+      className={`${isSelected ? "bg-opacity-0" : "bg-opacity-5"} ${paragraphStyle}`}
+    >
+      {label && <InputLabel label={label} id={id} />}
+      <textarea
+        id={id}
+        value={value}
+        rows={5}
+        onFocus={() => setIsSelected(true)}
+        onBlur={() => setIsSelected(false)}
+        placeholder={placeholder}
+        onChange={(e) => {
+          onChange(e);
+          setValueChanged(true);
+        }}
+        className={
+          valueStyle +
+          (valueChanged
+            ? " text-penni-text-regular-light-mode"
+            : " text-penni-text-tertiary-light-mode")
+        }
+      />
+    </div>
+  );
+}
+
+function DropdownMenu({ menuId, options, onChange }: DropdownMenuProps) {
+  return (
+    <div
+      className="absolute right-0 z-10 mx-4 -mt-3 flex w-56 origin-top-right flex-col rounded-penni-card bg-black bg-opacity-5 shadow-lg focus:outline-none"
```

Comment:
`bg-black bg-opacity-5` this opacity setting will make this div look not so good when it is on top of other components. And maybe we should consider how to design the UI when open the dropdown menu as it is ideally presented on the phone? Or improving it by making the width same as the dropdown button.
![image](https://github.com/codersforcauses/penni/assets/98962967/49e415f2-abc0-4730-8545-2f4d505ac09f)

Reply:
I think I left a comment for the opacity - yes `bg-opacity-5` is kind of a workaround. The background colour was not defined in Figma so we probably need to define it in the tailwind config file. The dropdown menu should probably extend the full width but I couldn't get it to work, might have another crack at it.

Reply:
Defined custom colour in tailwind config file. Dropdown menu is now full width (I will not disclose how long this took)